core-tools.markdown

```diff
@@ -0,0 +1,345 @@
+
+### Using external packages in ghci
+
+By default, ghci can only load and use packages that are [included with the GHC installation](https://downloads.haskell.org/ghc/latest/docs/html/users_guide/9.0.1-notes.html#included-libraries).
+
+However, users of [cabal-install](https://www.haskell.org/cabal) and [stack](http://haskellstack.org) can download and load external packages very easily using the following commands:
+
+Cabal-install:
+
+```sh
+cabal repl --build-depends random
+```
+
+Stack:
+
+```sh
+stack exec --package random -- ghci
+```
+
+And the modules of the relevant packages will be available for import:
+
+```sh
+GHCi, version 9.0.1: https://www.haskell.org/ghc/  :? for help
+λ> import System.Random
+λ> randomRIO (1, 10)
+1
+λ> randomRIO (1, 10)
+8
+λ> randomRIO (1, 10)
+5
+```
+
+Stack users can also use this feature with `runghc` and `ghc` by replacing `ghci` in the command above.
+
+Many more packages are waiting for you on [Hackage](https://hackage.haskell.org).
+
+## Compiling programs - Advanced usage
+
+In this section we'll cover more advanced usage available when compiling programs.
+Feel free to skip it!
+
+### Runtime options
+
+Compiling a Haskell program with the flag `-rtsopts` provides us with
+the ability to more finely control the Haskell runtime system when running the program.
+
+After compiling with `-rtsopts` with can pass flags to the Haskell runtime by
+listing them between `+RTS` and `-RTS` when invoking our Haskell program.
+For example, let's compile our `hello.hs` program with runtime options:
+
+```hs
+➜ ghc -rtsopts hello.hs 
+[1 of 1] Compiling Main             ( hello.hs, hello.o )
+Linking hello ...
+```
+
+And run it with `-sstderr` which will print some information
+about the runtime execution of the program: 
+
+```sh
+➜  ./hello +RTS -sstderr -RTS
+Hello, Haskell!
+          52,064 bytes allocated in the heap
+           3,312 bytes copied during GC
+          44,408 bytes maximum residency (1 sample(s))
+          25,224 bytes maximum slop
+               2 MiB total memory in use (0 MB lost due to fragmentation)
+
+                                     Tot time (elapsed)  Avg pause  Max pause
+  Gen  0         0 colls,     0 par    0.000s   0.000s     0.0000s    0.0000s
+  Gen  1         1 colls,     0 par    0.000s   0.000s     0.0001s    0.0001s
+
+  INIT    time    0.000s  (  0.000s elapsed)
+  MUT     time    0.000s  (  0.000s elapsed)
+  GC      time    0.000s  (  0.000s elapsed)
+  EXIT    time    0.000s  (  0.000s elapsed)
+  Total   time    0.000s  (  0.000s elapsed)
+
+  %GC     time       0.0%  (0.0% elapsed)
+
+  Alloc rate    717,569,877 bytes per MUT second
+
+  Productivity  27.4% of total user, 28.0% of total elapsed
+```
+
+This information is not very useful for such a trivial program,
+but as programs get more complex it will provide more insight
+regarding the runtime of the program.
+
+Alternatively, it is also possible to set the exact runtime flags
```

Comment:
I'd skip this fact entirely from this kind of first intro to the tooling. It takes bunch of space in the document but hardly brings much profit.